File: src/Budget.Domain/Entities/Account.cs
namespace Budget.Domain.Entities;

public class Account

{

    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public decimal StartingBalance { get; set; }

    public string UserId { get; set; } = string.Empty;

    public AccountVisibility Visibility { get; set; }
        = AccountVisibility.Personal;

    public decimal? SavingsGoalAmount { get; set; }

    public DateOnly? SavingsGoalDate { get; set; }

    public ICollection<Transaction> Transactions { get; set; }
        = new List<Transaction>();

}

File: frontend/src/types/account.ts
export type AccountVisibility = "Personal" | "Shared";

export type Account = {
    id: string;
    name: string;
    balance: number;
    startingBalance: number;
    visibility: AccountVisibility;
    savingsGoalAmount: number | null;
    savingsGoalDate: string | null;
    savingsGoalRemaining: number | null;
    requiredMonthlySavings: number | null;
};

export type CreateAccount = {
    name: string;
    startingBalance: number;
    visibility: AccountVisibility;
};

export type UpdateAccount = CreateAccount;

export type UpdateSavingsGoal = {
    targetAmount: number;
    targetDate: string;
};
